File: src/app/dashboard/tablegenerator/page.js
"use client";
import { useState } from "react";
import QRCard from '../../components/QRCard';
import {
  QrCode,
  Plus,
  ShieldCheck,
  Info,
  LayoutGrid,
  Zap
} from "lucide-react";
import apiConfig from "@/utils/apiConfig";
export default function TableGenerator() {
  const [count, setCount] = useState("");
  const [tables, setTables] = useState([]);
  const [loading, setLoading] = useState(false);

  
  const generate = async () => {
    if (!count || count <= 0) {
      return alert("Please enter a valid number of tables");
    }

    setLoading(true);

    try {
      const res = await fetch(
        `${apiConfig?.BASE_URL}/api/tables/generate`,
        {
          method: "POST",

          credentials: "include",   // 🔥 CRITICAL

          headers: {
            "Content-Type": "application/json",
          },

          body: JSON.stringify({
            tableCount: Number(count),

            // ❌ REMOVE THIS if backend uses req.user.restaurantId
            // restaurantId
          })
        }
      );

      const data = await res.json();

      if (!res.ok) {
        alert(data.message || "Failed to generate QR codes");
        return;
      }

      setTables(data.tables);

    } catch (err) {
      console.error("QR_GENERATE_ERROR:", err);
      alert("Network error");
    } finally {
      setLoading(false);
    }
  };
  return (
    <div className="min-h-screen bg-[#FAFAFA] p-6 md:p-12">
      <div className="max-w-6xl mx-auto">

        {/* Header */}
        <div className="mb-10 text-center md:text-left">
          <div className="inline-flex items-center gap-2 bg-orange-100 text-orange-600 px-4 py-1.5 rounded-full mb-4">
            <QrCode size={14} />
            <span className="text-[10px] font-black uppercase tracking-widest text-orange-600">Smart QR Generation</span>
          </div>
          <h1 className="text-4xl font-black text-gray-900 tracking-tight mb-2">Generate Table QRs</h1>
          <p className="text-gray-500 font-medium">Create instant digital menu access for every table in your restaurant.</p>
        </div>

        {/* Input Card */}
        <div className="bg-white rounded-[2.5rem] border border-gray-100 shadow-xl shadow-orange-100/20 p-8 md:p-10 mb-10">
          <div className="flex flex-col md:flex-row gap-6 items-end">
            <div className="flex-1 w-full">
              <label className="text-[10px] font-black text-gray-400 uppercase tracking-widest mb-3 block ml-1">How many tables?</label>
              <div className="relative group">
                <div className="absolute left-6 top-1/2 -translate-y-1/2 text-gray-400 group-focus-within:text-orange-600 transition-colors">
                  <LayoutGrid size={20} />
                </div>
                <input
                  type="number"
                  placeholder="e.g. 15"
                  value={count}
                  onChange={(e) => setCount(e.target.value)}
                  className="w-full pl-14 pr-6 py-5 rounded-2xl bg-gray-50 border-2 border-transparent focus:border-orange-500 focus:bg-white transition-all outline-none font-bold text-xl text-gray-800"
                />
              </div>
            </div>

            <button
              onClick={generate}
              disabled={loading}
              className="w-full md:w-auto bg-gray-900 text-white px-10 py-5 rounded-2xl font-black text-xs uppercase tracking-[0.2em] hover:bg-orange-600 transition-all shadow-xl shadow-gray-200 flex items-center justify-center gap-3 active:scale-[0.95] whitespace-nowrap"
            >
              {loading ? "Generating..." : <><Zap size={18} /> Generate QRs</>}
            </button>
          </div>

          <div className="mt-8 bg-blue-50 border border-blue-100 p-5 rounded-2xl flex gap-4">
            <Info className="text-blue-600 shrink-0 mt-1" size={18} />
            <p className="text-[11px] font-medium text-blue-700 leading-relaxed uppercase tracking-tight">
              <span className="font-black block mb-0.5">Trial Phase Benefit</span>
              Generating QR codes is <span className="font-bold underline">Free</span> in your trial. These codes will link directly to your digital menu.
            </p>
          </div>
        </div>

        {/* QR Grid */}
        {tables.length > 0 && (
          <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-6 animate-in fade-in slide-in-from-bottom-4 duration-500">
            {tables.map(t => (
              <QRCard key={t._id} table={t} />
            ))}
          </div>
        )}

        {/* Footer info */}
        <div className="mt-12 text-center text-[10px] font-bold text-gray-400 uppercase tracking-[0.2em] flex items-center justify-center gap-2">
          <ShieldCheck size={14} className="text-green-500" /> Powered by QRServe Secure Systems
        </div>
      </div>
    </div>
  );
}

File: lib/src/app/dashboard/tablegenerator/page.dart
import 'package:flutter/material.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';
import 'package:lucide_icons/lucide_icons.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../../utils/apiClient.dart';
import '../../context/AuthContext.dart';

class TableGeneratorPage extends ConsumerStatefulWidget {
  const TableGeneratorPage({super.key});
  @override
  ConsumerState<TableGeneratorPage> createState() => _TableGeneratorPageState();
}

class _TableGeneratorPageState extends ConsumerState<TableGeneratorPage> {
  String _count = '';
  List<dynamic> _tables = [];
  bool _isLoading = false;

  Future<void> _generate() async {
    final countInt = int.tryParse(_count);
    if (countInt == null || countInt <= 0) {
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(content: Text('Please enter a valid number of tables')),
      );
      return;
    }

    setState(() => _isLoading = true);

    try {
      final user = ref.read(authProvider).user;
      final restaurantId = user?['restaurantId'] ?? user?['_id'];

      final res = await apiFetch(
        '/api/tables/generate',
        method: 'POST',
        data: {
          'tableCount': countInt,
          if (restaurantId != null) 'restaurantId': restaurantId,
        },
      );
      if (res['tables'] != null) {
        setState(() => _tables = res['tables']);
      } else {
        if (mounted)
          ScaffoldMessenger.of(context).showSnackBar(
            SnackBar(
              content: Text(res['message'] ?? 'Failed to generate QR codes'),
            ),
          );
      }
    } catch (e) {
      if (mounted)
        ScaffoldMessenger.of(
          context,
        ).showSnackBar(SnackBar(content: Text('Error: $e')));
    } finally {
      if (mounted) setState(() => _isLoading = false);
    }
  }

  void _printQR(dynamic table) {
    ScaffoldMessenger.of(context).showSnackBar(
      SnackBar(
        content: Text(
          'Printing Table ${table['tableNumber']} (Requires native printing package)',
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFFFAFAFA),
      body: SingleChildScrollView(
        padding: EdgeInsets.all(24.r),
        child: Center(
          child: ConstrainedBox(
            constraints: BoxConstraints(maxWidth: 1100.w),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                // Header
                Align(
                  alignment: Alignment.centerLeft,
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Container(
                        padding: EdgeInsets.symmetric(
                          horizontal: 16.w,
                          vertical: 6.h,
                        ),
                        decoration: BoxDecoration(
                          color: Colors.orange.shade50,
                          borderRadius: BorderRadius.circular(20.r),
                        ),
                        child: Row(
                          mainAxisSize: MainAxisSize.min,
                          children: [
                            Icon(
                              LucideIcons.qrCode,
                              size: 14.sp,
                              color: Colors.orange,
                            ),
                            SizedBox(width: 8.w),
                            Text(
                              'SMART QR GENERATION',
                              style: TextStyle(
                                fontSize: 10.sp,
                                fontWeight: FontWeight.w900,
                                color: Colors.orange,
                                letterSpacing: 1,
                              ),
                            ),
                          ],
                        ),
                      ),
                      SizedBox(height: 16.h),
                      Text(
                        'Generate Table QRs',
                        style: TextStyle(
                          fontSize: 32.sp,
                          fontWeight: FontWeight.w900,
                          color: Color(0xFF0F172A),
                          letterSpacing: -1,
                        ),
                      ),
                      SizedBox(height: 8.h),
                      Text(
                        'Create instant digital menu access for every table in your restaurant.',
                        style: TextStyle(
                          fontSize: 14.sp,
                          fontWeight: FontWeight.bold,
                          color: Colors.grey,
                        ),
                      ),
                    ],
                  ),
                ),
                SizedBox(height: 40.h),

                // Input Card
                Container(
                  padding: EdgeInsets.all(32.r),
                  decoration: BoxDecoration(
                    color: Colors.white,
                    borderRadius: BorderRadius.circular(40.r),
                    border: Border.all(color: Colors.grey.shade100),
                    boxShadow: [
                      BoxShadow(
                        color: Colors.orange.shade50.withAlpha(50),
                        blurRadius: 20.r,
                      ),
                    ],
                  ),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      LayoutBuilder(
                        builder: (ctx, constraints) {
                          final isMobile = constraints.maxWidth < 600.w;
                          return Flex(
                            direction: isMobile
                                ? Axis.vertical
                                : Axis.horizontal,
                            crossAxisAlignment: CrossAxisAlignment.end,
                            children: [
                              Expanded(
                                flex: isMobile ? 0 : 1,
                                child: Column(
                                  crossAxisAlignment: CrossAxisAlignment.start,
                                  children: [
                                    Text(
                                      'HOW MANY TABLES?',
                                      style: TextStyle(
                                        fontSize: 10.sp,
                                        fontWeight: FontWeight.w900,
                                        color: Colors.grey,
                                        letterSpacing: 1,
                                      ),
                                    ),
                                    SizedBox(height: 12.h),
                                    Container(
                                      padding: EdgeInsets.symmetric(
                                        horizontal: 16.w,
                                      ),
                                      decoration: BoxDecoration(
                                        color: Colors.grey.shade50,
                                        borderRadius: BorderRadius.circular(
                                          16.r,
                                        ),
                                      ),
                                      child: Row(
                                        children: [
                                          Icon(
                                            LucideIcons.layoutGrid,
                                            size: 20.sp,
                                            color: Colors.grey,
                                          ),
                                          SizedBox(width: 12.w),
                                          Expanded(
                                            child: TextField(
                                              keyboardType:
                                                  TextInputType.number,
                                              onChanged: (v) => _count = v,
                                              decoration: InputDecoration(
                                                hintText: 'e.g. 15',
                                                border: InputBorder.none,
                                              ),
                                              style: TextStyle(
                                                fontSize: 20.sp,
                                                fontWeight: FontWeight.bold,
                                              ),
                                            ),
                                          ),
                                        ],
                                      ),
                                    ),
                                  ],
                                ),
                              ),
                              if (isMobile)
                                SizedBox(height: 16.h)
                              else
                                SizedBox(width: 24.w),
                              ElevatedButton.icon(
                                onPressed: _isLoading ? null : _generate,
                                icon: Icon(LucideIcons.zap, size: 18.sp),
                                label: Text(
                                  _isLoading ? 'GENERATING...' : 'GENERATE QRS',
                                  style: TextStyle(
                                    fontSize: 12.sp,
                                    fontWeight: FontWeight.w900,
                                    letterSpacing: 1,
                                  ),
                                ),
                                style: ElevatedButton.styleFrom(
                                  backgroundColor: const Color(0xFF0F172A),
                                  foregroundColor: Colors.white,
                                  padding: EdgeInsets.symmetric(
                                    horizontal: 40.w,
                                    vertical: 20.h,
                                  ),
                                  shape: RoundedRectangleBorder(
                                    borderRadius: BorderRadius.circular(16.r),
                                  ),
                                ),
                              ),
                            ],
                          );
                        },
                      ),
                      SizedBox(height: 32.h),
                      Container(
                        padding: EdgeInsets.all(20.sp),
                        decoration: BoxDecoration(
                          color: Colors.blue.shade50,
                          borderRadius: BorderRadius.circular(16.r),
                          border: Border.all(color: Colors.blue.shade100),
                        ),
                        child: Row(
                          crossAxisAlignment: CrossAxisAlignment.start,
                          children: [
                            Icon(
                              LucideIcons.info,
                              color: Colors.blue,
                              size: 18.sp,
                            ),
                            SizedBox(width: 16.w),
                            Expanded(
                              child: Column(
                                crossAxisAlignment: CrossAxisAlignment.start,
                                children: [
                                  Text(
                                    'TRIAL PHASE BENEFIT',
                                    style: TextStyle(
                                      fontSize: 11.sp,
                                      fontWeight: FontWeight.w900,
                                      color: Colors.blue,
                                      letterSpacing: 1,
                                    ),
                                  ),
                                  SizedBox(height: 4.h),
                                  Text(
                                    'Generating QR codes is FREE in your trial. These codes will link directly to your digital menu.',
                                    style: TextStyle(
                                      fontSize: 11.sp,
                                      fontWeight: FontWeight.bold,
                                      color: Colors.blue,
                                    ),
                                  ),
                                ],
                              ),
                            ),
                          ],
                        ),
                      ),
                    ],
                  ),
                ),
                SizedBox(height: 40.h),

                // QR Grid
                if (_tables.isNotEmpty)
                  ListView.builder(
                    shrinkWrap: true,
                    physics: const NeverScrollableScrollPhysics(),
                    itemCount: _tables.length,
                    itemBuilder: (ctx, idx) {
                      final table = _tables[idx];
                      return Container(
                        margin: EdgeInsets.only(bottom: 16.h),
                        padding: EdgeInsets.all(16.sp),
                        decoration: BoxDecoration(
                          color: Colors.white,
                          borderRadius: BorderRadius.circular(24.r),
                          border: Border.all(color: Colors.grey.shade100),
                          boxShadow: [
                            BoxShadow(
                              color: Colors.orange.shade50.withAlpha(20),
                              blurRadius: 10.r,
                              offset: Offset(0, 4.h),
                            ),
                          ],
                        ),
                        child: Column(
                          mainAxisSize: MainAxisSize.min,
                          crossAxisAlignment: CrossAxisAlignment.center,
                          children: [
                            // QR Code
                            Container(
                              padding: EdgeInsets.all(12.r),
                              decoration: BoxDecoration(
                                color: Colors.grey.shade50,
                                borderRadius: BorderRadius.circular(24.r),
                                border: Border.all(color: Colors.grey.shade100),
                              ),
                              child: Image.network(
                                'https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=${Uri.encodeComponent(table['qrUrl'] ?? '')}',
                                width: 140.w,
                                height: 140.w,
                                errorBuilder: (ctx, err, stack) => Icon(
                                  LucideIcons.qrCode,
                                  size: 140.w,
                                  color: Colors.black12,
                                ),
                              ),
                            ),
                            SizedBox(height: 24.h),
                            // Details
                            Column(
                              children: [
                                Text(
                                  'OUTLET ASSET',
                                  style: TextStyle(
                                    fontSize: 10.sp,
                                    fontWeight: FontWeight.w900,
                                    color: Colors.grey,
                                    letterSpacing: 2,
                                  ),
                                ),
                                SizedBox(height: 8.h),
                                Text(
                                  'Table ${table['tableNumber']}',
                                  style: TextStyle(
                                    fontSize: 28.sp,
                                    fontWeight: FontWeight.w900,
                                    color: const Color(0xFF0F172A),
                                    letterSpacing: -1,
                                  ),
                                ),
                              ],
                            ),
                            SizedBox(height: 24.h),
                            // Print Action
                            SizedBox(
                              width: double.infinity,
                              child: OutlinedButton.icon(
                                onPressed: () => _printQR(table),
                                icon: Icon(LucideIcons.printer, size: 18.sp),
                                label: Text(
                                  'PRINT LABEL',
                                  style: TextStyle(
                                    fontSize: 12.sp,
                                    fontWeight: FontWeight.w900,
                                    letterSpacing: 1,
                                  ),
                                ),
                                style: OutlinedButton.styleFrom(
                                  foregroundColor: const Color(0xFF0F172A),
                                  side: BorderSide(color: Colors.grey.shade200),
                                  backgroundColor: Colors.grey.shade50,
                                  padding: EdgeInsets.symmetric(vertical: 20.h),
                                  shape: RoundedRectangleBorder(
                                    borderRadius: BorderRadius.circular(16.r),
                                  ),
                                ),
                              ),
                            ),
                          ],
                        ),
                      );
                    },
                  ),

                SizedBox(height: 48.h),
                Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Icon(
                      LucideIcons.shieldCheck,
                      color: Colors.green,
                      size: 14.sp,
                    ),
                    SizedBox(width: 8.w),
                    Text(
                      'POWERED BY QRSERVE SECURE SYSTEMS',
                      style: TextStyle(
                        fontSize: 10.sp,
                        fontWeight: FontWeight.w900,
                        color: Colors.grey,
                        letterSpacing: 2,
                      ),
                    ),
                  ],
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
